src/test/java/com/internship/socialnetwork/service/impl/FriendshipServiceImplTest.java

```diff
@@ -0,0 +1,255 @@
+
+        // Verify that there are no more interactions with friendshipRepository
+        verifyNoMoreInteractions(friendshipRepository);
+    }
+
+    @Test
+    void shouldReturnFriendRequests_whenGetFriendRequests() {
+        // Given: Mocking new friendship
+        when(friendshipRepository.findAllFriendRequests(user.getId()))
+                .thenReturn(List.of(friendship));
+
+        // When: Getting friend requests
+        List<FriendshipDto> friendRequests = friendshipService.getFriendRequests(user.getId());
+
+        // Then: Verify that correct number of friend requests is returned
+        assertEquals(1, friendRequests.size());
+
+        // Verify that friendshipRepository.findAllFriendRequests() was called exactly once
+        verify(friendshipRepository, times(1)).findAllFriendRequests(user.getId());
+
+        // Verify that there are no more interactions with friendshipRepository
+        verifyNoMoreInteractions(friendshipRepository);
+    }
+
+    @Test
+    void shouldReturnSentFriendRequests_whenGetSentFriendRequests() {
+        // Given: Mocking new friendship
+        when(friendshipRepository.findAllSentFriendRequests(user.getId()))
+                .thenReturn(List.of(friendship));
+
+        // When: Getting friend requests
+        List<FriendshipDto> sentFriendRequests = friendshipService.getSentFriendRequests(user.getId());
+
+        // Then: Verify that correct number of sent friend requests is returned
+        assertEquals(1, sentFriendRequests.size());
+
+        // Verify that friendshipRepository.findAllSentFriendRequests() was called exactly once
+        verify(friendshipRepository, times(1)).findAllSentFriendRequests(user.getId());
+
+        // Verify that there are no more interactions with friendshipRepository
+        verifyNoMoreInteractions(friendshipRepository);
+
+    }
+
+    @Test
+    void shouldCreateFriendRequest_whenCreateFriendRequest_ifFriendshipDoesNotExist() {
+        // Given: Mocking new friendshipDto
+        FriendshipDto friendshipDto = convertToDto(friendship);
+
+        when(friendshipRepository.save(any(Friendship.class))).thenReturn(friendship);
+        when(userService.getUserModel(any(Long.class))).thenReturn(user);
+
+        // When: Creating new friend request
+        friendshipService.createFriendRequest(friendship.getUser().getId(), friendship.getFriend().getId());
+
+        // Then: Verify that friendshipDto is created correctly
+        assertEquals(friendshipDto.getUserId(), friendship.getUser().getId());
+        assertEquals(friendshipDto.getFriendId(), friendship.getFriend().getId());
+        assertEquals(friendshipDto.getFriendUsername(), friendship.getFriend().getUsername());
+        assertEquals(friendshipDto.getUsername(), friendship.getUser().getUsername());
+
+        // Verify that friendshipRepository.findFriendship(...) is called exactly once
+        verify(friendshipRepository, times(1)).findFriendship(any(Long.class), any(Long.class));
```

Comment:
getUserModel should also be verified, whatever you mock with when() should go to verify() too